docs/dev/release_checklist.md

```diff
@@ -0,0 +1,132 @@
+# Release Checklist
+
+This document is intended for app maintainers and covers the steps to perform when releasing new versions of the app.
+
+## Minor Version Bumps
+
+### Update Requirements
+
+Every minor version release should refresh `poetry.lock`, so that it lists the most recent stable release of each package. To do this:
+
+0. Run `poetry update --dry-run` to have Poetry automatically tell you what package updates are available and the versions it would upgrade to. This requires an existing environment created from the lock file (i.e. via `poetry install`).
+1. Review each requirement's release notes for any breaking or otherwise noteworthy changes.
+2. Run `poetry update <package>` to update the package versions in `poetry.lock` as appropriate.
+3. If a required package requires updating to a new release not covered in the version constraints for a package as defined in `pyproject.toml`, (e.g. `Django ~3.1.7` would never install `Django >=4.0.0`), update it manually in `pyproject.toml`.
+4. Run `poetry install` to install the refreshed versions of all required packages.
+5. Run all tests (`poetry run invoke tests`) and check that the UI and API function as expected.
+
+### Update Documentation
+
+Commit any resulting changes from the following sections to the documentation before proceeding with the release.
+
+!!! tip
+    Fire up the documentation server in your development environment with `poetry run mkdocs serve`! This allows you to view the documentation site locally and automatically rebuilds it as you make changes.
```

Comment:
Can we add the url here?

Reply:
```
INFO    -  [15:22:24] Serving on http://127.0.0.1:8001/projects/dev-example/en/latest/
```

This one? When you run the serve command it gives you the URL it listens on.

Reply:
I've added a note there about the URL being in the command output. You have to run it anyway before you can open the URL.